### ISSUE TITLE
IWYU includes the wrong header for alias templates

### ISSUE THREAD

**Issue by det** [include-what-you-use/include-what-you-use#703]
Reduced test case:

test.cc:
```
#include "foo.hh"
#include <string>

int main() {
  Foo<std::string> foo;
}
```

foo.hh:
```
#pragma once

#include "bar.hh"

template <typename T>
using Foo = Bar<T, int>;
```

bar.hh:
```
#pragma once

template<typename T, typename U>
class Bar {
};
```

IWYU reports:
```
test.cc should add these lines:
#include "bar.hh"  // for Bar

test.cc should remove these lines:
- #include "foo.hh"  // lines 1-1

The full include-list for test.cc:
#include <string>  // for string
#include "bar.hh"  // for Bar
---
```

But following this advice would break compilation as Foo would no longer be found.

**Comment by kimgr** [comment]
Thanks! Yes, IWYU is still most confident in the C++03 space, so language features from C++11 and onwards will probably confuse it. I wonder if the case here is that the alias template is modelled in the AST, but we look through it as part of template analysis.

This needs to be investigated, and your reduced repro should make a good testcase.

**Comment by jbytheway** [comment]
I happened upon the same problem, and have a proposed fix in #772.
